src/client/helpers/api-movie-post.js

```diff
@@ -0,0 +1,37 @@
+import { getMovies } from "./api-get";
+
+const port = import.meta.env.VITE_PORT;
+const apiUrl = `http://localhost:${port}`;
+const token = localStorage.getItem("token")
+
+const buildUrl = () => `${apiUrl}/movie`
+
+const buildOptionsObject = (form) => {
+    const options = {
+        method: "POST", 
+        headers: {
+            "content-type":"application/json", 
+            "Authorization": token,
+        }, 
+        body: JSON.stringify(form)
+    }
+    return options
+}
+
+const handleResponse = (data, updateList) => {
+        if(data.error) {
+            window.alert(data.error)
```

Comment:
This should probably throw an error instead of doing the alert. It's nice that you've split the fetching code for movies into it's own file, but by using `window.alert` you're mixing functional behavior (fetching data) with display behavior (displaying the alert)

Reply:
so then, something along the lines of 
`  if(data.error) {
   throw new Error(data.error)`
in this file, and then in MyMoviesList.jsx have something like:

`     const renderMyMovies = () => {
        try {
            getMyMovies(setMyMovies)
        } catch (e) {
            window.alert(e)
        }
    } `
? 

I had a very brief go at a refactor to see if I could get that to work - the error got thrown, but went uncaught.  I'll probably have another look at it tomorrow.